File: 03Builder/Students/2019/JarzembinskiBartlomiej/builder/computer_builder.py
from abc import ABC, abstractmethod

class ComputerBuilder(ABC):

    def get_computer(self):
        return self.computer

    @abstractmethod
    def build_case(self):
        pass

    @abstractmethod
    def build_motherboard(self):
        pass

    @abstractmethod
    def build_power_supply(self):
        pass

    @abstractmethod
    def build_cpu(self):
        pass

    @abstractmethod
    def build_gpu(self):
        pass

    @abstractmethod
    def build_drive(self):
        pass

    @abstractmethod
    def build_ram(self):
        pass

File: 03Builder/Students/2019/KaczmarzTomasz/BuilderPattern/ComputerBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderPattern
{
    abstract class ComputerBuilder
    {
        protected Computer computer;

        Computer Computer { get { return computer; } }

        public abstract ComputerBuilder BuildCPU();
        public abstract ComputerBuilder BuildGPU();
        public abstract ComputerBuilder BuildMotherboard();
        public abstract ComputerBuilder BuildRAM();

        public static implicit operator Computer(ComputerBuilder builder)
        {
            return builder.BuildCPU().BuildGPU().BuildMotherboard().BuildRAM().Computer;
        }
    }
}
